utils/compression/compression.ts

```diff
@@ -0,0 +1,14 @@
+import LZString from "lz-string";
+import { DecompressedData } from "../../src/store";
+
+export const compress = (data: object): string => {
+  return LZString.compressToEncodedURIComponent(JSON.stringify(data));
```

Comment:
How do we handle scenarios where the compressed string is larger than 2MB?

- Proceed regardless, and let the browser handle it (i.e. crash the browser tab for some browsers)?
- Alert the user and disable link-sharing
- Proceed, but warn the user
- Something else?

Reply:
Thanks @mttrbrts, thought of this too!
Currently, our implementation does not include handling for scenarios where the compressed string exceeds the 2MB limit(i.e like 10,000's of lines of code scenerio). Given this, the behavior would likely be: Proceed regardless, and let the browser handle it (i.e., crash the browser tab for some browsers perhaps): At the moment, I believe our code will proceed regardless of the compressed string size.

Acknowledging the importance of addressing this, and it's definitely something we could prioritize in an upcoming Pull Requests. Maybe a solution to alert the user and disable link-sharing if the size limit is exceeded, or providing a warning from your suggestions